src/middlewared/middlewared/plugins/pool.py

```diff
@@ -735,6 +735,9 @@ def add(attr):
             await self.middleware.call(
                 'notifier.change_dataset_share_type', id, data['share_type'].lower()
             )
+        elif data['type'] == 'VOLUME' and 'volsize' in data:
```

Comment:
We need to compare old data and new data here. New UI will always send `volsize` in `data`, even unchanged, won't it?

Reply:
Thats a GUI issue, it should not send volsize if not changed.